nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -392,17 +455,41 @@ def diarize(self, paths2audio_files: List[str] = None, batch_size: int = 0):
 
         self.AUDIO_RTTM_MAP = audio_rttm_map(self._diarizer_params.manifest_filepath)
 
+        out_rttm_dir = os.path.join(self._out_dir, 'pred_rttms')
+        os.makedirs(out_rttm_dir, exist_ok=True)
+
         # Speech Activity Detection
         self._perform_speech_activity_detection()
 
         # Segmentation
-        self._run_segmentation()
+        self.multi_scale_dict = self.parse_scale_configs(**self._diarizer_params.speaker_embeddings.parameters)
+        if self.multi_scale_dict:
 
-        # Embedding Extraction
-        self._extract_embeddings(self.subsegments_manifest_path)
+            # Multi-scale Segmentation
+            multi_scale_emb_ts_spkrs = {}
+            for scale_idx, (time_length, shift_length) in self.multi_scale_dict['scale_dict'].items():
 
-        out_rttm_dir = os.path.join(self._out_dir, 'pred_rttms')
-        os.makedirs(out_rttm_dir, exist_ok=True)
+                # Segmentation for the current scale (scale_idx)
+                self._run_segmentation(window=time_length, shift=shift_length, scale_tag=f'_scale{scale_idx}')
+                logging.info(
+                    f"Extracting multiscale embeddings for Diarization: scale-{scale_idx}, {self.subsegments_manifest_path}"
+                )
+
+                # Embedding Extraction for the current scale (scale_idx)
+                self._extract_embeddings(self.subsegments_manifest_path, scale_tag=f'_scale{scale_idx}')
+
+                multi_scale_emb_ts_spkrs[scale_idx] = [self.embeddings, self.time_stamps]
+
+            self.embeddings, self.time_stamps, multi_scale_data = get_multiscale_time_stamps(
+                multi_scale_emb_ts_spkrs, self.multi_scale_dict
+            )
+        else:
+            # Single-scale Segmentation
+            self._run_segmentation()
+
+            # Single-scale Embedding Extraction
+            self._extract_embeddings(self.subsegments_manifest_path)
+            multi_scale_data = None
```

Comment:
My suggestion is to have a loop without if case and parse window and shift lengths accordingly for extracting embeddings and mapping timestamps. 
(More detailed discussion offline)

Reply:
This comment is reflected in the latest commit.